vertx-core/src/main/java/io/vertx/core/impl/deployment/DefaultDeploymentManager.java

```diff
@@ -36,7 +36,7 @@ public DefaultDeploymentManager(VertxImpl vertx) {
   }
 
   private String generateDeploymentID() {
-    return UUID.randomUUID().toString();
+    return Long.valueOf(System.nanoTime()).toString();
```

Comment:
This needs to be globally unique when running in clustered mode with HA enabled

So perhaps something like:

```java
    if (vertx.isClustered() && vertx.haManager()!=null) {
      return UUID.randomUUID().toString();
    }
    // Use a counter?
```

Reply:
👌

Reply:
Updated

Reply:
It's pretty common to deploy verticles concurrently. Even when Vert.x is not clustered, the returned value should be unique.

Reply:
?

Reply:
I have updated it to use Random, is that what you meant?

Reply:
I meant incrementing an `AtomicLong` counter instead of using a random value (uniqueness is guaranteed and it shouldn't change the perf results you got)